ruby/ql/lib/codeql/ruby/security/UnsafeDeserializationCustomizations.qll

```diff
@@ -223,13 +223,109 @@ module UnsafeDeserialization {
           // anywhere to set the default options to a known safe mode.
           not ojLoad.hasExplicitKnownMode(_) and
           not exists(SetOjDefaultOptionsCall setOpts |
-            setOpts.getValue().(OjOptionsHashWithModeKey).hasSafeMode()
+            setOpts.getValue().(OptionsHashWithModeKey).hasKnownMode(getAKnownOjModeName(true))
           )
         )
       )
     }
   }
 
+  /**
+   * The first argument in a call to `Oj.object_load`, always considered as a
+   * sink for unsafe deserialization. (global and local mode options are ignored)
+   */
+  class OjObjectLoadArgument extends Sink {
+    OjObjectLoadArgument() {
+      this = API::getTopLevelMember("Oj").getAMethodCall("object_load").getArgument(0)
+    }
+  }
+
+  /**
+   * Unsafe deserialization utilizing the Ox gem
+   * See: https://github.com/ohler55/ox
+   */
+  private string getAKnownOxModeName(boolean isSafe) {
+    result = ["generic", "limited", "hash", "hash_no_attrs"] and isSafe = true
+    or
+    result = "object" and isSafe = false
+  }
+
+  /**
+   * A call node that sets `Ox.default_options`.
+   *
+   * ```rb
+   * Ox.default_options = { mode: :limited, effort: :tolerant }
+   * ```
+   */
+  private class SetOxDefaultOptionsCall extends DataFlow::CallNode {
+    SetOxDefaultOptionsCall() {
+      this = API::getTopLevelMember("Ox").getAMethodCall("default_options=")
+    }
+
+    /**
+     * Gets the value being assigned to `Ox.default_options`.
+     */
+    DataFlow::Node getValue() {
+      result.asExpr() =
+        this.getArgument(0).asExpr().(CfgNodes::ExprNodes::AssignExprCfgNode).getRhs()
```

Comment:
Shouldn't this just be as follows? I believe the first argument of a `default_options=` function is the right-hand side of the assignment already. Same for the `Oj` variant of this class.

```suggestion
      result =  this.getArgument(0)
```

```suggestion
      result = this.getArgument(0)
```

Reply:
Yes, now that you say it... makes a lot of sense 😄 

(FYI: I'll update this branch and check what fails)